pkg/xds/sync/dataplane_proxy_builder.go

```diff
@@ -111,7 +114,25 @@ func (p *DataplaneProxyBuilder) resolveVIPOutbounds(meshContext xds_context.Mesh
 	if dataplane.Spec.Networking.GetTransparentProxying() == nil {
 		newOutbounds := []*core_xds.Outbound{}
 		for _, o := range dataplane.Spec.Networking.Outbound {
-			newOutbounds = append(newOutbounds, &core_xds.Outbound{LegacyOutbound: o})
+			if o.BackendRef != nil {
+				newOutbounds = append(newOutbounds, &core_xds.Outbound{
```

Comment:
Maybe we can extract this into a new method?

Reply:
I created `AsOutbounds()` method for MeshService, MeshExternalService, MeshMultiZoneService, Dataplane and it simplified the code for mesh and dpp context builders.